src/client/datascience/notebook/intellisense/jupyterCompletionProvider.ts

```diff
@@ -157,9 +192,43 @@ export class JupyterCompletionProvider implements CompletionItemProvider {
             metadata: {}
         };
     }
+
+    private async getPylanceCompletions(
+        document: TextDocument,
+        position: Position,
+        context: CompletionContext,
+        cancelToken: CancellationToken
+    ) {
+        if (document.notebook) {
+            const client = await this.languageClientProvider.getLanguageClient(document.notebook);
+            if (client) {
+                // Use provider so it gets translated by middleware
+                const feature = client.getFeature(lsp.CompletionRequest.method);
+                const provider = feature.getProvider(document);
+                if (provider) {
+                    const results = await provider.provideCompletionItems(document, position, cancelToken, context);
+                    if (results && 'items' in results) {
+                        return results.items;
+                    } else {
+                        return results;
+                    }
+                }
+            }
+        }
+    }
+}
+
+function positionInsideString(word: string, position: Position) {
+    const indexDoubleQuote = word.indexOf('"');
+    const indexSingleQuote = word.indexOf("'");
```

Comment:
We might want to exclude `"""` (docstrings)

Reply:
Skipping this for now. Doc strings are not going to work the same (as they are likely on a different line)

Reply:
The default is no completions for strings anyway.